controller/volume_controller.go

```diff
@@ -848,18 +848,40 @@ func (vc *VolumeController) cleanupExtraHealthyReplicas(v *longhorn.Volume, e *l
 }
 
 func (vc *VolumeController) cleanupEvictionRequestedReplicas(v *longhorn.Volume, rs map[string]*longhorn.Replica) (bool, error) {
+	log := getLoggerForVolume(vc.logger, v)
+
+	// If there is no non-evicting healthy replica,
+	// Longhorn should retain one evicting healthy replica.
+	hasNonEvictingHealthyReplicas := false
+	evictingHealthyReplica := ""
 	for _, r := range rs {
-		if r.Status.EvictionRequested {
-			if err := vc.deleteReplica(r, rs); err != nil {
-				vc.eventRecorder.Eventf(v, v1.EventTypeWarning,
-					EventReasonFailedEviction,
-					"volume %v failed to evict replica %v",
-					v.Name, r.Name)
-				return false, err
-			}
-			logrus.Debugf("Evicted replica %v", r.Name)
-			return true, nil
+		if !datastore.IsAvailableHealthyReplica(r) {
+			continue
+		}
+		if !r.Status.EvictionRequested {
+			hasNonEvictingHealthyReplicas = true
+			break
 		}
+		evictingHealthyReplica = r.Name
+	}
+
+	for _, r := range rs {
+		if !r.Status.EvictionRequested {
+			continue
+		}
+		if !hasNonEvictingHealthyReplicas && r.Name == evictingHealthyReplica {
+			log.Debugf("Cannot evicting replica %v for now since there is no other healthy replica", r.Name)
```

Comment:
Should we use `log.Warnf` here?

Reply:
```suggestion
			log.Debugf("Cannot evicting replica %v for now since there is no other healthy replica", r.Name)
```

			log.Warnf("Cannot evict replica %v for now since there is no other healthy replica", r.Name)

Reply:
Sure we can use `Warnf`